src/batch/src/executor/iceberg_scan.rs

```diff
@@ -0,0 +1,185 @@
+///
+/// #[tokio::test]
+/// async fn test_iceberg_scan() {
+///     let iceberg_scan_executor = IcebergScanExecutor {
+///         database_name: "demo_db".into(),
+///         table_name: "demo_table".into(),
+///         file_selector: FileSelector::select_all(),
+///         iceberg_config: IcebergConfig {
+///             database_name: "demo_db".into(),
+///             table_name: "demo_table".into(),
+///             catalog_type: Some("storage".into()),
+///             path: "s3a://hummock001/".into(),
+///             endpoint: Some("http://127.0.0.1:9301".into()),
+///             access_key: "hummockadmin".into(),
+///             secret_key: "hummockadmin".into(),
+///             region: Some("us-east-1".into()),
+///             ..Default::default()
+///         },
+///         snapshot_id: None,
+///         batch_size: 1024,
+///         schema: Schema::new(vec![
+///             Field::with_name(DataType::Int64, "seq_id"),
+///             Field::with_name(DataType::Int64, "user_id"),
+///             Field::with_name(DataType::Varchar, "user_name"),
+///         ]),
+///         identity: "iceberg_scan".into(),
+///     };
+///
+///     let stream = Box::new(iceberg_scan_executor).execute();
+///     #[for_await]
+///     for chunk in stream {
+///         let chunk = chunk.unwrap();
+///         println!("{:?}", chunk);
+///     }
+/// }
+/// ```
+
+pub struct IcebergScanExecutor {
+    database_name: String,
+    table_name: String,
+    iceberg_config: IcebergConfig,
+    snapshot_id: Option<i64>,
+    file_selector: FileSelector,
+    batch_size: usize,
+
+    schema: Schema,
+    identity: String,
+}
+
+impl Executor for IcebergScanExecutor {
+    fn schema(&self) -> &risingwave_common::catalog::Schema {
+        &self.schema
+    }
+
+    fn identity(&self) -> &str {
+        &self.identity
+    }
+
+    fn execute(self: Box<Self>) -> super::BoxedDataChunkStream {
+        self.do_execute().boxed()
+    }
+}
+
+impl IcebergScanExecutor {
+    #[try_stream(ok = DataChunk, error = BatchError)]
+    async fn do_execute(self: Box<Self>) {
+        let catalog = self.iceberg_config.create_catalog().await?;
+
+        let table_ident = TableIdentifier::new(vec![self.database_name, self.table_name]).unwrap();
+        let table = catalog
+            .load_table(&table_ident)
+            .await
+            .map_err(BatchError::Iceberg)?;
+
+        let table_scan: TableScan = table
+            .new_scan_builder()
+            .with_snapshot_id(
+                self.snapshot_id
+                    .unwrap_or_else(|| table.current_table_metadata().current_snapshot_id.unwrap()),
+            )
+            .with_batch_size(self.batch_size)
+            .with_column_names(self.schema.names())
+            .build()
+            .map_err(|e| BatchError::Internal(anyhow!(e)))?;
+        let file_scan_stream: icelake::io::FileScanStream =
+            table_scan.scan(&table).await.map_err(BatchError::Iceberg)?;
```

Comment:
Usually we do this in planning phase, so that we only need to read metadata for only once, and the scan executor only needs to contains file names.

Reply:
wondering if we can do the same as long as setting `snapshot_id`

Reply:
However, icelake currently doesn't provide a way to read Files directly. If we have a better API later, we can avoid such planning in the future.

Reply:
> wondering if we can do the same as long as setting snapshot_id

Yes, in fact we should set `snapshot_id`, otherwise different tasks may see different snapshots.

Reply:
cc @ZENOTME  Could you help to add this api in icelake?

Reply:
I would suggest to pending this to wait for icelake's api support, WDYT? cc @chenzl25

Reply:
I intend to move fast and present a demo version. Any optimization could be done in parallel as well. 😄

Reply:
Cool, let's move

Reply:
> cc @ZENOTME Could you help to add this api in icelake?

Sorry for replying late. Yes. And I agree to move it now. For the new API, let's track it in icelake as a new feature.